### ISSUE TITLE
Use more diverse bugs in the duplicate classifier

### ISSUE THREAD

**Issue by marco-c** [mozilla/bugbug#541]
I think this is why we are seeing artificially high results.

Since we limit the number of non-duplicate bugs, because of the way we are generating the couples (combination of first bug ID with all other bug IDs, then second bug ID with all other bug IDs, and so on) we end up generating couples where the first element is always the same. Indeed, when we do the combination of the first bug ID with all other bug IDs, we already reach the total number of labels we wanted.

Adding this snippet in the code right after we add the labels (setting `classes = {}` before each step):
```Python
        from collections import Counter
        all_ids_selected = list(sum(([bug_id1, bug_id2] for bug_id1, bug_id2 in classes.keys()), []))
        print(Counter(all_ids_selected).most_common(24))
        print(f"Unique bugs: {len(set(all_ids_selected))}")
```
I see for purely duplicates:
```
[(1390125, 27), (1435472, 22), (1332714, 19), (1412559, 17), (1373490, 15), (1370401, 13), (1386222, 13), (1387340, 13), (1345661, 12), (1460221, 12), (1324399, 11), (1372041, 11), (1389484, 11), (1324255, 10), (1350277, 10), (1392633, 10), (1393213, 10), (1397278, 10), (1330552, 9), (1348432, 9), (1360392, 9), (1363372, 9), (1377950, 9), (1389476, 9)]
Unique bugs: 11493
```
For hybrid:
```
[(1376257, 3500), (1310752, 1), (35, 1), (101, 1), (1310831, 1), (786819, 1), (1049215, 1), (1311731, 1), (1311790, 1), (1311797, 1), (1311830, 1), (1049816, 1), (1312165, 1), (1050244, 1), (1312587, 1), (1312588, 1), (788319, 1), (1312610, 1), (1312611, 1), (1312613, 1), (1312619, 1), (1312621, 1), (1313019, 1), (1146776, 1)]
Unique bugs: 3501
```
For purely non-duplicates:
```
[(1310752, 3500), (35, 1), (101, 1), (1310831, 1), (786819, 1), (1049215, 1), (1311731, 1), (1311790, 1), (1311797, 1), (1311830, 1), (1049816, 1), (1312165, 1), (1050244, 1), (1312587, 1), (1312588, 1), (788319, 1), (1312610, 1), (1312611, 1), (1312613, 1), (1312619, 1), (1312621, 1), (1313019, 1), (1146776, 1), (1313272, 1)]
Unique bugs: 3501
```

So basically, for duplicate bugs we are good, we already have diverse bugs.
For non-duplicates instead, we are overfitting as all our labels either contain 1376257 or 1310752.

**Comment by ayush-1506** [comment]
Oh, I see. One way to get around this would be to have a counter (initially set to a small value, say, 10) and generating pairs up to this counter. So for the hybrid/non-duplicate case, the first bug will have 10 cases, 2nd bug will have 10 and so on. 
@marco-c ?

**Comment by marco-c** [comment]
The best option would be to generate combinations randomly.
